### ISSUE TITLE
Remote aiming 

### ISSUE THREAD

**Issue by jamesol1** [DerPavlov/Cannons#27]
Hi Pav,
Would there be a way you could allow for remote aiming through right clicking a sign with say a clock to enter aiming mode, left click to fire.. providing one is using automatic reloading. I think this would be a really neat feature for those that use the plugin alongside movecraft.

Movecraft allows you to remotely aim and fire vanilla cannons from a 'bridge', this would allow you to do the same with cannons (which has a lot more variety with projectiles/settings).

Many thanks

**Comment by DerPavlov** [comment]
I think it is easier for Movecraft to implement these changes, since Cannons only handles one cannon at a time and Movecraft knows the cannons on a craft.
I could only give you a list of cannons you can control or your cannons in the vicinity.
Actually I like the idea of firing a full broadside.

**Comment by jamesol1** [comment]
> I think it is easier for Movecraft to implement these changes, since Cannons only handles one cannon at a time and Movecraft knows the cannons on a craft.
> I could only give you a list of cannons you can control or your cannons in the vicinity.
> Actually I like the idea of firing a full broadside.

That would work great. It would probably have to be the cannons you own in your vicinity. If there was a way of allowing another person to use your cannons for the remote aiming that would be good!

I'm imagining a sign 'Remote Cannons' then listing the names below of people who can also remote aim them. Not sure if that would work but just an idea. Or you could make it command based /cannons remoteaccess add (player name).
/cannons remoteaccess remove (player name).
So players on your team can also fire too.

Yeah i think it could add a new dynamic to cannons.

**Comment by DerPavlov** [comment]
You can set other people as operators for your cannons, but you have to to it for every single cannon at the moment.
I have to think about how it will fit in the current code, but it will be a ton of work to implement it.
I could add a tag to the cannons to set it to remote access and I there is a function to find cannons close to you, but that would be not player specific.
In other words a cannon set to remote access would be accessible for all player in the vicinity with can operate this cannon.
Maybe it is easier to make a list of cannons for every player, so everybody can be in charge for their cannons.

**Comment by jamesol1** [comment]
> Maybe it is easier to make a list of cannons for every player, so everybody can be in charge for their cannons.

Yeah that would work too actually. People could setup their own cannons to aim on a ship for example. I'm assuming the aiming would still work in the same way as it would on a moving craft (aslong as the isSignRequired is enabled).

**Comment by DerPavlov** [comment]
At the moment I am very busy with other projects, I don't believe this will come anytime soon. If you have somebody which can do the coding, that would be really helpful.

**Comment by jamesol1** [comment]
> At the moment I am very busy with other projects, I don't believe this will come anytime soon. If you have somebody which can do the coding, that would be really helpful.

Ah i see, well thanks for getting back to me. Unfortunetly i don't. I guess i'd just say whenever you fancy adding it, that would be cool but it's totally up to you, thanks for your work on the plugin.

**Comment by DerPavlov** [comment]
I thought a little bit about the problem and the easiest solution might be a command to toggle /areaaimingmode. That can probably done in a few hours.
You could also use a sign to jump into this mode, but it is on until you deactivate it again via sign or command.
However, since you now control more than one cannon it will need some changes.

**Comment by jamesol1** [comment]
> I thought a little bit about the problem and the easiest solution might be a command to toggle /areaaimingmode. That can probably done in a few hours.
> You could also use a sign to jump into this mode, but it is on until you deactivate it again via sign or command.
> However, since you now control more than one cannon it will need some changes.

Any sort of work around would be great. I suppose you'd probably want to make it so you could only aim a group of cannons aiming in the same direction. Feel free to do this how you wish, anything that could make it come into fruition through it being quicker to do would be great.

**Comment by jamesol1** [comment]
I wonder if you could make it so you could simply right click cannons one owns with x tool item (like how you do with the auto aiming - clock) but then select other cannons you own to then aim and fire at the same time. You might need to allow for a custom/further aiming away from the cannon distance but yeah.

Would that be easier? If it is it would fix issues with cannons aiming in different directions of players can manually decide which cannons to collectively and remotely aim/fire.

**Comment by Pvprod34** [comment]
This would work great for my server too actually. Pavlov have you looked into it yet any chance? Not to push you on it or anything.
